File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxp = 0
        l,r = 0, 1
        while r<len(prices):
            if prices[l]>prices[r]:
                l=r
            else:
                maxp = max(maxp, prices[r]-prices[l])
            r+=1
        return maxp   

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    maxSlidingWindow(nums, k) {
        let res = []
        for (let i=0;i<nums.length;i++){
            // if (i+k!=nums.length){
            //     res.push(Math.max(...nums.slice(i,i+k)))
            // }
            if (i+k<=nums.length)
            res.push(Math.max(...nums.slice(i,i+k)))
        }
        return res
    }
}
